docs/smart-contracts/meebits_erc721.sol

```diff
@@ -0,0 +1,679 @@
+    function setApprovalForAll(address operator, bool _approved) external;
+    function isApprovedForAll(address owner, address operator) external view returns (bool);
+    function safeTransferFrom(address from, address to, uint256 tokenId, bytes calldata data) external;
+}
+
+/**
+ * Minimal interface to Cryptopunks for verifying ownership during Community Grant.
+ */
+interface Cryptopunks {
+    function punkIndexToAddress(uint index) external view returns(address);
+}
+
+interface ERC721TokenReceiver
+{
+    function onERC721Received(address _operator, address _from, uint256 _tokenId, bytes calldata _data) external returns(bytes4);
+}
+
+library SafeMath {
+
+    /**
+    * @dev Multiplies two numbers, throws on overflow.
+    */
+    function mul(uint256 a, uint256 b) internal pure returns (uint256 c) {
+        if (a == 0) {
+            return 0;
+        }
+        c = a * b;
+        require(c / a == b);
+        return c;
+    }
+
+    /**
+    * @dev Integer division of two numbers, truncating the quotient.
+    */
+    function div(uint256 a, uint256 b) internal pure returns (uint256) {
+        // assert(b > 0); // Solidity automatically throws when dividing by 0
+        // uint256 c = a / b;
+        // assert(a == b * c + a % b); // There is no case in which this doesn't hold
+        return a / b;
+    }
+
+    /**
+    * @dev Subtracts two numbers, throws on overflow (i.e. if subtrahend is greater than minuend).
+    */
+    function sub(uint256 a, uint256 b) internal pure returns (uint256) {
+        require(b <= a);
+        return a - b;
+    }
+
+    /**
+    * @dev Adds two numbers, throws on overflow.
+    */
+    function add(uint256 a, uint256 b) internal pure returns (uint256 c) {
+        c = a + b;
+        require(c >= a);
+        return c;
+    }
+}
+
+contract Meebits is IERC721 {
+
+    using SafeMath for uint256;
+
+    /**
+     * Event emitted when minting a new NFT. "createdVia" is the index of the Cryptopunk/Autoglyph that was used to mint, or 0 if not applicable.
+     */
+    event Mint(uint indexed index, address indexed minter, uint createdVia);
+
+    /**
+     * Event emitted when a trade is executed.
+     */
+    event Trade(bytes32 indexed hash, address indexed maker, address taker, uint makerWei, uint[] makerIds, uint takerWei, uint[] takerIds);
+
+    /**
+     * Event emitted when ETH is deposited into the contract.
+     */
+    event Deposit(address indexed account, uint amount);
+
+    /**
+     * Event emitted when ETH is withdrawn from the contract.
+     */
+    event Withdraw(address indexed account, uint amount);
+
+    /**
+     * Event emitted when a trade offer is cancelled.
+     */
+    event OfferCancelled(bytes32 hash);
+
+    /**
+     * Event emitted when the public sale begins.
+     */
+    event SaleBegins();
+
+    /**
+     * Event emitted when the community grant period ends.
+     */
+    event CommunityGrantEnds();
```

Comment:
Mint, Trade, Deposit, Withdraw, OfferCancelled, SaleBegins, CommunityGrantEnds are all events in addition to ERC-721.